### PR TITLE
ISPN-11943 Single MediaType ignored in JSON config

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.redhat.com/browse/ISPN-11943

### CODE REVIEW

File: core/src/main/java/org/infinispan/configuration/cache/EncodingConfiguration.java [hunk 1]
```diff
@@ -18,17 +20,30 @@
  */
 public final class EncodingConfiguration implements Matchable<EncodingConfiguration>, ConfigurationInfo {
 
+   public static final AttributeDefinition<String> MEDIA_TYPE = AttributeDefinition.builder("mediaType", null, String.class).build();
+   private final AttributeSet attributes;
+
    private ContentTypeConfiguration keyDataType, valueDataType;
```
Comment (anistor) [inline core/src/main/java/org/infinispan/configuration/cache/EncodingConfiguration.java:26]: These could be final

File: core/src/main/java/org/infinispan/configuration/cache/EncodingConfiguration.java [hunk 2]
```diff
@@ -18,17 +20,30 @@
...
+
+   public EncodingConfiguration(AttributeSet attributes, ContentTypeConfiguration keyDataType, ContentTypeConfiguration valueDataType) {
+      this.attributes = attributes;
       this.keyDataType = keyDataType;
       this.valueDataType = valueDataType;
       this.contentTypeConfigurations = Arrays.asList(keyDataType, valueDataType);
    }
 
+   @Override
+   public AttributeSet attributes() {
```
Comment (anistor) [inline core/src/main/java/org/infinispan/configuration/cache/EncodingConfiguration.java:43]: I see that the toString, hashCode and equals are not based on ```attributes``` and do not include the newly added attribute.

File: core/src/main/java/org/infinispan/configuration/cache/EncodingConfiguration.java [hunk 3]
```diff
@@ -18,17 +20,30 @@
  */
 public final class EncodingConfiguration implements Matchable<EncodingConfiguration>, ConfigurationInfo {
 
+   public static final AttributeDefinition<String> MEDIA_TYPE = AttributeDefinition.builder("mediaType", null, String.class).build();
+   private final AttributeSet attributes;
+
    private ContentTypeConfiguration keyDataType, valueDataType;
    private final List<ConfigurationInfo> contentTypeConfigurations;
 
    public static final ElementDefinition ELEMENT_DEFINITION = new DefaultElementDefinition(ENCODING.getLocalName());
```
Comment (anistor) [inline core/src/main/java/org/infinispan/configuration/cache/EncodingConfiguration.java:29]: this can be package local

**Comment by gustavonalle** [comment]
Updated.

**Comment by anistor** [comment]
Merged. Thanks @gustavonalle !

**Pull request closed without merging**